### PR TITLE
23 add docker support

### LINKED ISSUE #23: Add docker support
Spinning this off a docker image will ease a lot of setup and deployment efforts. I don't have to install node and npm stuff on my host machine. Would help hugely with teams when it comes to development collaboration. Alas, but it's working on my machine!

I opened a [PR](https://github.com/netlify/victor-hugo/pull/25) a few days ago in a similar project but the maintainers find it not aligning with the project's goal of keeping it simple( which I understand ). Reading through the comments on #18 it seems there's an interest in adding docker support. Let me know if you welcome a PR for this. I'll open one right away.

### PR DESCRIPTION
**Pull request by eyedol**
**- Summary**
- Adds a script in the scripts folder that creates a docker container, builds it and runs.
- Adds a folder called docker which contains docker template files.
- Adds a script in the scripts folder that creates hugulp required folders.
- Updates `README` with clear instructions on how to run the script.

It fixes #23 

**- Test plan**
Run command below to create a default docker container. Note, the script is already executable.
```bash
./scripts/create-docker-machine-and-run-it
```

### CODE REVIEW

File: scripts/create-docker-machine-and-run-it [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+  sed -i '' -e "s|^FROM node:\([0-9\.]*\).*|FROM node:$NODE_VERSION|" \
+    Dockerfile
+  # Replace hugo version with set value
+  sed -i '' -e "s|^ARG HUGO_VERSION=\([0-9\.]*\).*|ARG HUGO_VERSION=$HUGO_VERSION|" \
+    Dockerfile
+  # Replace app home directory with set value
+  sed -i '' -e "s|\$HOME/[a-zA-Z0-9]*/$|\$HOME/$APP_NAME/|g" \
+    Dockerfile
+  # Replace website name with set value
+  sed -i '' -e "s|APP_NAME=website|APP_NAME=$APP_NAME|" \
```
Comment (jbrodriguez) [inline scripts/create-docker-machine-and-run-it]: Should docker/Dockerfile.template contain an ```APP_NAME=website``` line ?

File: scripts/create-hugo-site [hunk 2]
```diff
@@ -0,0 +1,80 @@
+#!/bin/sh
+#
+# Creates pre-requiste hugo site
+set -eu
+
+readonly PROGNAME=$(basename "$0")
+readonly ARGS="$@"
+APP_NAME=
```
Comment (jbrodriguez) [inline scripts/create-hugo-site]: Shouldn't this be ```APP_NAME=app``` ?

**Comment by jbrodriguez** [comment]
Hi @eyedol, it looks great !!

I made two comments, just to make sure, I haven't done any Docker development in quite a while :)

If it's ok as it is, I'll merge the PR.

Thinking ahead, would you consider becoming a collaborator ?

I wouldn't mind moving the project over to a github organization (hugulp?), so that you and other people could get involved.

File: scripts/create-docker-machine-and-run-it (hunk 1 above)
Comment (eyedol) [reply scripts/create-docker-machine-and-run-it]: It shouldn't contain that. Left over from a refactor

File: scripts/create-hugo-site (hunk 2 above)
Comment (eyedol) [reply scripts/create-hugo-site]: I'm forcing the user to provide that via the command line option. Thinking more about it, I'll make it `APP_NAME=app` and relax the app name option.

**Comment by eyedol** [comment]
@jbrodriguez Thanks for the review. I have addressed your comments in my latest push. I squashed everything into a single commit to prevent clutter in the commit history.

Yeah, I'm happy to contribute whenever I can. Feel free to add me as a collaborator. Making it an org will be better I think. Easier to track people. Feel free to add me on there when you have it created.

**Comment by jbrodriguez** [comment]
@eyedol awesome, thanks for the PR !

I'll work on the org next week.

**Pull request merged**